src/mode.js

```diff
@@ -121,7 +121,8 @@ function getToken(stream, state) {
       }
 
       // Otherwise, match a Terminal.
-      if (expected.match && expected.match(token)) {
+      const invariant = expected.constraints && expected.constraints(token);
```

Comment:
Should this be `!expected.constraints || expected.constraints(token);` ?

I think it would be clearer if `constraints` was a function that returns true when is passes constraints, rather than true when it fails them.